### PR TITLE
meson: update to version 1.4.0.

### PR DESCRIPTION
**Pull request by SpidFightFR**
#### Testing the changes
- I tested the changes in this PR: **NO**

### CODE REVIEW

**Comment by SpidFightFR** [comment]
I have yet to test apps using `ymir`, since i'm not familiar with that tool yet, i might take a while before making this PR ready.

**Comment by SpidFightFR** [comment]
So i tested out ymir, here are the results of my first run

<details><summary>ymir todo grab -f build_style:meson (WARNING: Big text wall)</summary>

```

good:
        AppStream
        GPaste
        Komikku
        Kooha
        MangoHud
        Minder
        Mousai
        NetworkManager
        PackageKit
        SwayNotificationCenter
        The-Powder-Toy
        Tuba
        Waybar
        a2jmidid
        accountsservice
        adwaita-icon-theme
        aisleriot
        alsa_rnnoise
        amtk
        apostrophe
        appstream-glib
        aravis
        arc-theme
        at-spi2-core
        atkmm
        atomix
        audacious
        audacious-plugins
        babl
        balsa
        baobab
        basu
        blanket
        bolt
        bookworm
        brisk-menu
        bubblewrap
        budgie-screensaver
        caerbannog
        cage
        cagebreak
        cairo
        cairomm
        cairomm1.16
        cantarell-fonts
        cglm
        chayang
        cheese
        cinnamon
        cinnamon-control-center
        cinnamon-desktop
        cinnamon-menus
        cinnamon-session
        cinnamon-settings-daemon
        cjs
        clipped
        colord
        colord-gtk
        contractor
        cups-pk-helper
        cyanrip
        d-feet
        dav1d
        dconf
        ddgtk
        deja-dup
        desktop-file-utils
        devhelp
        diffuse
        dmenu-wayland
        dosbox-staging
        dpdk
        drm_info
        easyeffects
        efl
        elogind
        enlightenment
        eog-plugins
        eolie
        epiphany
        evince
        extension-manager
        faba-icon-theme
        fastd
        fbida
        fcft
        file-roller
        flatpak
        flatpak-builder
        fnott
        folks
        fontmanager
        foot
        fprintd
        fractal
        fragments
        fsearch
        fuse-sshfs
        fuse3
        fwupd
        fwupd-efi
        gamehub
        gamemode
        ganv
        gcab
        gcolor3
        gcr
        gcr4
        gdk-pixbuf
        gdk-pixbuf-xlib
        gdm
        geary
        gedit
        gedit-plugins
        geeqie
        gegl
        geoclue2
        geocode-glib
        gfeeds
        ghex
        girara
        gitg
        gjs
        glade3
        glib
        glibmm
        glibmm2.68
        glu
        gnome-2048
        gnome-authenticator
        gnome-autoar
        gnome-backgrounds
        gnome-bluetooth
        gnome-bluetooth1
        gnome-browser-connector
        gnome-builder
        gnome-calculator
        gnome-calendar
        gnome-characters
        gnome-chess
        gnome-clocks
        gnome-color-manager
        gnome-contacts
        gnome-control-center
        gnome-desktop
        gnome-dictionary
        gnome-disk-utility
        gnome-font-viewer
        gnome-initial-setup
        gnome-klotski
        gnome-maps
        gnome-mines
        gnome-multi-writer
        gnome-music
        gnome-nettool
        gnome-nibbles
        gnome-online-accounts
        gnome-passbook
        gnome-podcasts
        gnome-power-manager
        gnome-remote-desktop
        gnome-robots
        gnome-screenshot
        gnome-secrets
        gnome-settings-daemon
        gnome-shell
        gnome-shell-extensions
        gnome-sound-recorder
        gnome-sudoku
        gnome-system-monitor
        gnome-taquin
        gnome-tetravex
        gnome-text-editor
        gnome-tour
        gnome-usage
        gnome-video-effects
        gnote
        gom
        granite
        granite7
        graphene
        greybird-themes
        grilo
        grilo-plugins
        gsound
        gssdp
        gssdp1.2
        gst-libav
        gst-omx
        gst-plugins-base1
        gst-plugins-good1
        gst-plugins-ugly1
        gst-rtsp-server
        gst1-python3
        gstreamer1
        gthumb
        gtk+3
        gtk-doc
        gtk-layer-shell
        gtk-vnc
        gtk4
        gtkmm
        gtksourceview4
        gtksourceview5
        gtranslator
        gucharmap
        gupnp
        gupnp-av
        gupnp-dlna
        gupnp-igd
        gupnp1.2
        gupnp1.2-igd
        gwe
        hexchat
        hitori
        i3
        i3lock
        i3status
        iagno
        igt-gpu-tools
        iio-sensor-proxy
        inih
        io.elementary.files
        io.elementary.icons
        io.elementary.music
        io.elementary.photos
        io.elementary.print
        io.elementary.stylesheet
        io.elementary.terminal
        io.elementary.videos
        iputils
        iwgtk
        jalv
        jo
        json-glib
        jsoncpp
        jsonrpc-glib
        kanshi
        kiwix-lib
        kiwix-tools
        kmscon
        knot-resolver
        labwc
        lavalauncher
        lc0
        lcms2
        lemoa
        libaccounts-glib
        libadwaita
        libchamplain
        libcloudproviders
        libdecor
        libdex
        libdisplay-info
        libdrm
        libei
        libepoxy
        libfprint
        libfprint0
        libgdata
        libgepub
        libgit2-glib
        libglvnd
        libgnomekbd
        libgnt
        libgudev
        libgusb
        libgweather
        libhandy
        libhandy1
        libinput
        libinput-debug-gui
        libjcat
        liblc3
        libmanette
        libmicrodns
        libmpdclient
        libnice
        libnotify
        libnvme
        libosinfo
        libpanel
        libpciaccess
        libpeas
        libplacebo
        libportal
        libratbag
        librist
        libsecret
        libshumate
        libsigc++3
        libsixel
        libslirp
        libsoup
        libsoup3
        libva
        libva-intel-driver
        libva-utils
        libvdpau
        libvirt
        libvirt-glib
        libwacom
        libwnck
        libwpe
        libxcvt
        libxkbcommon
        libxmlb
        libzim
        light-locker
        lightdm-webkit2-greeter
        lightsoff
        lighttpd
        linux-driver-management
        lite-xl
        loksh
        lollypop
        lxc
        lxcfs
        marco
        mate-desktop
        mate-media
        mate-polkit
        mate-system-monitor
        megapixels
        megatools
        meld
        mesa
        mm-common
        mozo
        mpc
        mpvpaper
        mu4e
        muffin
        muon
        mutter
        nautilus
        ncmpc
        neatvnc
        nemo
        nemo-fileroller
        nemo-image-converter
        nemo-preview
        nemo-python
        network-manager-applet
        newsflash
        noise-repellent
        notes-up
        nutty
        nvidia-vaapi-driver
        nvme-cli
        nwg-launchers
        orc
        osinfo-db-tools
        p11-kit
        pacman
        pamixer
        pangomm
        pangomm2.48
        pantheon-agent-polkit
        pantheon-screenshot
        paper-icon-theme
        paprefs
        paps
        patchage
        pdftag
        peek
        phodav
        pinentry-bemenu
        piper
        pipewire
        pithos
        pixman
        playerctl
        plocate
        polari
        polkit
        polychromatic
        power-profiles-daemon
        poweralertd
        pscircle
        pulseaudio
        pulseaudio-equalizer-ladspa
        pulseeffects-legacy
        purple-plugin-pack
        pystring
        quadrapassel
        quickDocs
        radare2
        rage-player
        razergenie
        retro-gtk
        rizin
        rubberband
        seahorse
        seahorse-nautilus
        seatd
        sequeler
        serd
        setzer
        shared-mime-info
        simple-scan
        slurp
        snes9x-gtk
        sonic-visualiser
        sord
        sound-juicer
        spice-protocol
        srain
        sratom
        sushi
        sway
        swaybg
        swayfx
        swayidle
        swaykbdd
        swaylock
        swell-foop
        switchboard-plug-bluetooth
        switchboard-plug-printers
        sysprof
        system-monitoring-center
        taisei
        tali
        tau
        template-glib
        tepl
        terminology
        timeshift
        tio
        tllist
        tofi
        tootle
        totem
        totem-pl-parser
        tracker
        tracker-miners
        transmission-remote-gtk
        umockdev
        upower
        usbredir
        vala-language-server
        vala-panel
        vala-panel-appmenu
        valabind
        viewnior
        vips
        vkBasalt
        vkQuake
        vmaf
        vte3
        wayfire-plugins-extra
        wayland
        wayland-protocols
        wayland-utils
        waypipe
        wayvnc
        webp-pixbuf-loader
        webrtc-audio-processing
        weston
        wf-config
        wf-recorder
        wf-shell
        wike
        wl-clipboard
        wlclock
        wlogout
        wlr-randr
        wlrctl
        wlroots0.15
        wlsunset
        wmenu
        wob
        wofi
        wpebackend-fdo
        wtype
        xapps
        xarcan
        xdg-dbus-proxy
        xdg-desktop-portal
        xdg-desktop-portal-gtk
        xdg-desktop-portal-wlr
        xf86-input-wacom
        xfce4-alsa-plugin
        xkeyboard-config
        xorg-server
        xorg-server-xwayland
        yambar
        yaru
        yelp-tools
        zathura-cb
        zathura-djvu
        zathura-pdf-mupdf
        zathura-pdf-poppler
        zathura-ps
        zchunk
        zenity
        zim-tools
bad:
        Endeavour
        Marker
        amberol
        aml
        avizo
        bijiben
        blueprint-compiler
        budgie-control-center
        budgie-desktop
        budgie-desktop-view
        celluloid
        cinnamon-screensaver
        citations
        coeurl
        corecollector
        cozy
        d-spy
        dconf-editor
        dinit
        eog
        foliate
        four-in-a-row
        fuzzel
        gamescope
        gdm-settings
        geocode-glib-libsoup2
        glib-networking
        glmark2
        gnome-boxes
        gnome-console
        gnome-mahjongg
        gnome-photos
        gnome-session
        gnome-software
        gnome-terminal
        gnome-tweaks
        gnome-weather
        gobject-introspection
        goodvibes
        grim
        gsettings-desktop-schemas
        gst-plugins-bad1
        gst1-editing-services
        gstreamer-vaapi
        gtkgreet
        gtkmm4
        gupnp-tools
        gvfs
        gxml
        harfbuzz
        imv
        io.elementary.calculator
        io.elementary.calendar
        io.elementary.code
        jack_mixer
        janet
        libcacard
        libdazzle
        libgedit-amtk
        libgedit-gtksourceview
        libgexiv2
        libgnome-games-support
        libgxps
        libliftoff
        libmediaart
        libnma
        libpeas2
        libqrtr-glib
        libscfg
        libsigc++
        libsrtp
        libva-glx
        libvarlink
        lutris
        lv2
        mako
        mate-power-manager
        mda-lv2
        mpd
        mpdscribble
        mpv
        nautilus-python
        pango
        picom
        python3-cairo
        python3-gobject
        python3-syndom
        python3-xapp
        rest
        rtaudio
        rygel
        scrcpy
        shortwave
        shotwell
        simde
        snapshot
        spice-gtk
        sqsh-tools
        swappy
        swayimg
        switchboard
        switchboard-plug-network
        systemd-boot-efistub
        tecla
        text-engine
        tilix
        virglrenderer
        virt-viewer
        wayfire
        wbg
        wcm
        wdisplays
        wireplumber
        wlroots0.16
        wlroots0.17
        xdg-desktop-portal-gnome
        xorgproto
        xprintidle
        xreader
        zathura
        zix
bad_checksum:
        mpdevil
        planner
bad_fetch:
        five-or-more
        gnome-connections
        patchmatrix
        pax-utils
unsupported_arch:
        rand48_r
```
</details>

I guess imma try and see why things don't work out on some apps... If someone wanna help, please go ahead !

**Comment by SpidFightFR** [comment]
I'm unsure why but after git pulling the void-packages repo and doing nothing else, a lot more packages worked...

<details><summary>ymir todo grab -f build_style:meson (WARNING: Big text wall)</summary>

```
bad:
        budgie-control-center
        budgie-desktop-view
        mpd
        switchboard-plug-network
        wayfire
bad_checksum:
        fuzzel
        mpdevil
        planner
bad_fetch:
        patchmatrix
        pax-utils
        virt-viewer
unsupported_arch:
        rand48_r
```
</details>
(the missing packages in the list are good)

I'll see what i can do for now.

**Comment by SpidFightFR** [comment]
Here are the remaining packages:
<details><summary>ymir todo grab -f build_style:meson (WARNING: Big text wall)</summary>

```
bad:
        budgie-desktop-view
        mpd
        switchboard-plug-network
        wayfire

```
</details>

If someone comes up with a patch or something, please share.

**Comment by SpidFightFR** [comment]
Alright lads, i think meson 1.4.0 is near to be ready on void, here is the current todo list:
- budgie-desktop-view
- switchboard-plug-network
- wayfire

I'll see if i can patch this in a near future.

**Comment by SpidFightFR** [comment]
Alright, i made a few more additions to the commit, to fix Wayfire - respectively:

- new package: nlohmann-json - a brand new dependency to wayfire after update - I just noted i forgot to introduce properly the MIT licence, i'm going to fix this ASAP.
- wf-config: update to version 0.8.0. - in order to fix wayfire building error after update.
- wayfire: update to version 0.8.1 - in order to fix it's own building issue

To all of the above packages here are the mandatory checks:

#### Testing the changes
- I tested the changes in this PR: **Briefely**

#### Local build testing
- I built this PR locally for my native architecture, (x64-GlibC)

**Comment by SpidFightFR** [comment]
There ! And as a bonus, i now know how to edit a commit i made before other commits ! :smile: 
I see this as an absolute win !

**Comment by SpidFightFR** [comment]
I'm actually confused about the checks for nlohmann-json, the folder should be found, the install went well on my xbps-src docker...

If someone has an idea, i'm all ears, thanks !

**Comment by SpidFightFR** [comment]
perfect, right at the moment where wlroots got updated lol.

That's alright, i just updated switchboard-plug-network. Now builds for me...

**Comment by SpidFightFR** [comment]
I'm currently stuck with budgie-desktop-view, here are the build [logs](https://gist.github.com/SpidFightFR/4334993fd69c5faa4ebc77230ee2dad6).

**Comment by meator** [comment]
Related to https://github.com/BuddiesOfBudgie/budgie-desktop-view/issues/29

**Comment by SpidFightFR** [comment]
Alright, so we checked with @meator in the IRC:

budgie-desktop-view is currently BROKEN in upstream, i disabled the error checks but upstream seems stuck back in time... Kinda annoying.

**Comment by SpidFightFR** [comment]
Either we leave the current build of budgie-desktop-view as such (unchanged), knowing that we can't build it again untill upstream fix their work. In that case, the PR is ready.

Or we go with the "disabling error checking" thing. Not a huge fan of it, not gonna lie.

Also can we re-run a `check build` please ? It seems those are broken cuz i made the last commit right when wlroots was updated...

**Comment by meator** [comment]
> Also can we re-run a `check build` please ? It seems those are broken cuz i made the last commit right when wlroots was updated...

You can rerun it by force pushing. If you have nothing to push, you can run

```
git commit --amend --no-edit
```

on a clean repo. It will not change any files, but it will change the SHA of the last commit, which will make it force pushable.

**Comment by SpidFightFR** [comment]
> > Also can we re-run a `check build` please ? It seems those are broken cuz i made the last commit right when wlroots was updated...
> 
> You can rerun it by force pushing. If you have nothing to push, you can run
> 
> ```
> git commit --amend --no-edit
> ```
> 
> on a clean repo. It will not change any files, but it will change the SHA of the last commit, which will make it force pushable.

thanks for the tip, learnt something new today :pray:

**Comment by meator** [comment]
Another solution would be to add `-D werror=false` to Meson arguments. Thanks ztrawhcse for that suggestion.

**Comment by eli-schwartz** [comment]
No problem, happy to help. Removing `-w` is kind of an obscure issue to have, lol...

**Comment by SpidFightFR** [comment]
Thanks to the both of you for the advice, it is now added to the template file.

**Comment by SpidFightFR** [comment]
should i change the comment in the template file of `budgie-desktop-view` or leave it that way ?

File: srcpkgs/wayfire/template [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+revision=1
+_utils_commit=08553c418f164bf5e84613d27447a32e380b75f0
 _touch_commit=8974eb0f6a65464b63dd03b842795cb441fb6403
 create_wrksrc=yes
 build_wrksrc="${pkgname}-${version}"
 build_style=meson
 configure_args="-Dprint_trace=false"
 hostmakedepends="pkg-config wayland-devel"
-makedepends="wf-config-devel wlroots0.16-devel cairo-devel pango-devel
+makedepends="wf-config-devel wlroots-devel cairo-devel pango-devel json-c++
```
Comment (meator) [inline srcpkgs/wayfire/template:12]: You've changed `wlroots0.16-devel` to `wlroots-devel`.

Comment (meator) [reply srcpkgs/wayfire/template:12]: Don't do that.

Comment (SpidFightFR) [reply srcpkgs/wayfire/template]: My apologies, it should be fixed now.

File: srcpkgs/budgie-desktop-view/template [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # Template file for 'budgie-desktop-view'
 pkgname=budgie-desktop-view
 version=1.3
-revision=1
+revision=2
 build_style=meson
+configure_args="-D werror=false" #Dirty fix to budgie-desktop-view not compiling on meson 1.4.0 due to upstream errors
```
Comment (ahesford) [inline srcpkgs/budgie-desktop-view/template:6]: This doesn't warrant a comment, just specify the argument.

File: srcpkgs/wayfire/template (hunk 1 above)
Comment (meator) [reply srcpkgs/wayfire/template]: I am mistaken it looks like. A different solution is necessary.

File: srcpkgs/budgie-desktop-view/template (hunk 2 above)
Comment (SpidFightFR) [reply srcpkgs/budgie-desktop-view/template]: Alrighty, removed it.

**Comment by SpidFightFR** [comment]
Thank you for your patience, guys, i'm still learning a lot of new things about Void-Linux and xbps-src, i appreciate the time and efforts your putting in to help me and my goofs... :sweat_smile: :pray:

**Comment by SpidFightFR** [comment]
It is finished.
The PR is finally ready (?)

File: srcpkgs/wayfire/template [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Template file for 'wayfire'
 pkgname=wayfire
-version=0.7.5
-revision=2
-_utils_commit=889fdafa7b09fc6ce70313fd3f067cfc8c21a2ac
+version=0.8.1
+revision=1
+_utils_commit=08553c418f164bf5e84613d27447a32e380b75f0
```
Comment (meator) [inline srcpkgs/wayfire/template:5]: This commit hash is incorrect; it points to HEAD, not to the version wayfire 0.8.1 expects. If you have questions, I've created a mini-tutorial explaining exactly this problem: https://xbps-src-tutorials.github.io/mini-tutors/git-submodules.html It explains it from a template creation standpoint, but most of it also applies here when updating templates.

```suggestion
_utils_commit=15f8e16721585ae3eaf278ba71d7064237eb23f5
```

Comment (SpidFightFR) [reply srcpkgs/wayfire/template:5]: neat, thank you !
Saving this one for the future !

**Comment by ahesford** [comment]
I ran through some extensive, but not exhaustive, randomized builds across all supported architectures and the following packages have failed:

    modified:   srcpkgs/AppStream/template
    modified:   srcpkgs/bijiben/template
    modified:   srcpkgs/fractal/template
    modified:   srcpkgs/mesa/template
    modified:   srcpkgs/nemo-preview/template
    modified:   srcpkgs/pax-utils/template
    modified:   srcpkgs/planner/template
    modified:   srcpkgs/virt-viewer/template

I do not yet know why these failed and will look more closely soon.

**Comment by SpidFightFR** [comment]
> I ran through some extensive, but not exhaustive, randomized builds across all supported architectures and the following packages have failed:
> 
> ```
> modified:   srcpkgs/AppStream/template
> modified:   srcpkgs/bijiben/template
> modified:   srcpkgs/fractal/template
> modified:   srcpkgs/mesa/template
> modified:   srcpkgs/nemo-preview/template
> modified:   srcpkgs/pax-utils/template
> modified:   srcpkgs/planner/template
> modified:   srcpkgs/virt-viewer/template
> ```
> 
> I do not yet know why these failed and will look more closely soon.

working on it atm. looks like bijiben is fixed, for some reasons updating meson introduced makedeps lacking bugs...

**Comment by SpidFightFR** [comment]
What a mess... I hope the commit history length isn't an issue...!

**Comment by SpidFightFR** [comment]
Argh... I'll check those errors when I got a few hours free.

**Comment by meator** [comment]
I wonder if there are any other programs that are affected by the vala issue caused by meson 1.4.0. But `budgie-desktop-view` was in a pretty unique situation, because it set `Werror`. It might be still worth testing meson vala packages on `x86_64-musl` or something.

<details><summary>Templates containing "vala" and "meson"</summary>
<p>

```
AppStream
GPaste
Minder
NetworkManager
PackageKit
SwayNotificationCenter
Tuba
avizo
babl
baobab
bookworm
bookworm
budgie-desktop-view
budgie-desktop
cheese
clipped
colord-gtk
colord
contractor
dconf-editor
dconf
deja-dup
five-or-more
folks
fontmanager
fontmanager
four-in-a-row
fwupd
gamehub
gcab
gcr
gcr4
geary
geary
gedit-plugins
gegl
geoclue2
gitg
gnome-2048
gnome-boxes
gnome-builder
gnome-calculator
gnome-chess
gnome-clocks
gnome-connections
gnome-contacts
gnome-control-center
gnome-initial-setup
gnome-klotski
gnome-mahjongg
gnome-mines
gnome-nibbles
gnome-online-accounts
gnome-robots
gnome-sudoku
gnome-taquin
gnome-tetravex
gnome-usage
granite
granite7
grilo
grilo
gsound
gssdp
gssdp1.2
gtk-layer-shell
gtk-vnc
gtksourceview4
gtksourceview5
gupnp-av
gupnp-dlna
gupnp
gupnp1.2
gxml
iagno
io.elementary.calculator
io.elementary.calendar
io.elementary.calendar
io.elementary.calendar
io.elementary.calendar
io.elementary.code
io.elementary.files
io.elementary.music
io.elementary.photos
io.elementary.photos
io.elementary.print
io.elementary.terminal
io.elementary.videos
jsonrpc-glib
lavalauncher
libaccounts-glib
libadwaita
libchamplain
libcloudproviders
libdazzle
libdex
libgdata
libgepub
libgexiv2
libgit2-glib
libgnome-games-support
libgudev
libgusb
libgweather
libhandy
libhandy1
libjcat
libmanette
libmediaart
libnma
libosinfo
libosinfo
libpanel
libpeas
libpeas2
libportal
libsecret
libshumate
libsoup
libsoup3
libvirt-glib
libvirt-glib
lightsoff
meson
notes-up
nutty
nutty
pantheon-agent-polkit
pantheon-screenshot
pdftag
peek
planner
pulseaudio
quadrapassel
quickDocs
retro-gtk
rygel
seahorse
seahorse
sequeler
shotwell
simple-scan
spice-gtk
swell-foop
switchboard-plug-bluetooth
switchboard-plug-network
switchboard-plug-printers
switchboard
template-glib
timeshift
tootle
tracker
uhttpmock
umockdev
vala-language-server
vala-panel-appmenu
vala-panel
valabind
valabind
vte3
xapps
xfce4-alsa-plugin
```

</p>
</details>

**Comment by SpidFightFR** [comment]
@meator i tried to look at the error of pax-utils... I'm quite confused of what that can be... Do you have any ideas on your side? (I'll take more time to check tomorrow either ways)

**Comment by SpidFightFR** [comment]
> I ran through some extensive, but not exhaustive, randomized builds across all supported architectures and the following packages have failed:
> 
> ```
> modified:   srcpkgs/AppStream/template
> modified:   srcpkgs/bijiben/template
> modified:   srcpkgs/fractal/template
> modified:   srcpkgs/mesa/template
> modified:   srcpkgs/nemo-preview/template
> modified:   srcpkgs/pax-utils/template
> modified:   srcpkgs/planner/template
> modified:   srcpkgs/virt-viewer/template
> ```
> 
> I do not yet know why these failed and will look more closely soon.

Hey, so i checked in further details...
For pax-utils:
The current template is broken not because of meson but because of a restructuring in upstream.
the distfile location changed, i would suggest to use [this link](https://gitweb.gentoo.org/proj/pax-utils.git/snapshot/pax-utils-1.3.7.tar.gz) instead (this is the link to the archive with the proper tag, which can be found here: https://gitweb.gentoo.org/proj/pax-utils.git/tag/?h=v1.3.7).

Tests do fail as of now with pax-utils. I struggle to find a reason why, but from what i found: `lddtree -l /bin/bash` command fails to execute, and this causes the test itself to fail. I can't seem to find if it's related to meson or if it's caused by something else.

**Comment by leahneukirchen** [comment]
fixed in 2f5d8827d57

**Comment by SpidFightFR** [comment]
> fixed in [2f5d882](https://github.com/void-linux/void-packages/commit/2f5d8827d5770b060a515b2db366f61a1d878548)

nice ! did you managed to build it tho ? cuz the check process failed for me, i'm kinda curious... :sweat_smile:

**Comment by eli-schwartz** [comment]
See https://gitweb.gentoo.org/repo/gentoo.git/tree/app-misc/pax-utils/pax-utils-1.3.7.ebuild#n21

The correct download site was never distfiles.gentoo.org (that is a mirror network, distfiles can expire from there). If you do want to use it anyway, the subdirectory structure used is described at https://www.gentoo.org/glep/glep-0075.html

**Comment by SpidFightFR** [comment]
`gtksourceview5` is now broken, i didn't find a solution yet.

**Comment by meator** [comment]
> `gtksourceview5` is now broken, i didn't find a solution yet.

Wouldn't you know it, Vala strikes again. I've called it, I knew that `budgie-desktop-view` isn't the only package affected by this.

**Comment by SpidFightFR** [comment]
> > `gtksourceview5` is now broken, i didn't find a solution yet.
> 
> Wouldn't you know it, Vala strikes again. I've called it, I knew that `budgie-desktop-view` isn't the only package affected by this.

hey, sorry i brought some update in the IRC, but the TLDR is:

apparently updating the package solves the issue, only problem is, the `gtk_doc` build option is deprecated for `documentation` instead, moving from `gtk_doc` to `gi-docgen` from what i understood.

I'm searching solutions but it might take time.

**Comment by SpidFightFR** [comment]
Wait hold up a minute, look:

<details><summary>Error-build.log</summary>

```
FAILED: gtksourceview/gtksourceview-5.vapi 
/usr/bin/vapigen --quiet --library=gtksourceview-5 --directory=/builddir/gtksourceview5-5.10.0/build/gtksourceview --pkg=gdk-pixbuf-2.0 --pkg=gio-2.0 --pkg=gtk4 --pkg=pango --pkg=cairo --metadatadir=/builddir/gtksourceview5-5.10.0/gtksourceview /builddir/gtksourceview5-5.10.0/build/gtksourceview/GtkSource-5.gir
GtkSource-5.gir:18586.7-18591.84: warning: Signal `GtkSource.View.smart_home_end' conflicts with property of the same name
GtkSource-5.gir:9812.7-9812.43: error: The type name `Gtk.AccessibleText' could not be found
GtkSource-5.gir:17230.7-17230.43: error: The type name `Gtk.AccessibleText' could not be found

```
</details>

Something ain't right with vapi...

**Comment by SpidFightFR** [comment]
You are correct, it IS vala.

**Comment by SpidFightFR** [comment]
Tests so far seem good...
Does anyone have an issue with meson and bulk building?

**Comment by SpidFightFR** [comment]
Hello @meator , vala has been updated, is the issue you mentioned for a handful of package solved or is it still relevant?

**Comment by SpidFightFR** [comment]
> > Hello @meator , vala has been updated, is the issue you mentioned for a handful of package solved or is it still relevant?
> 
> I believe it's still relevant, it is an issue with Meson and not with Vala itself, but it looks like only `budgie-desktop-view` is affected because of its `Werror`.
> 
> You said that you were bulk building Meson packages. How did that go?

all is good, my second run had errors but it's ymir glitching out for reasons i don't know... I re-ran all failed packages individually and everything is good on my side.

File: srcpkgs/fractal/template [hunk 4]
```diff
@@ -8,7 +8,7 @@ hostmakedepends="cargo pkg-config glib-devel gettext"
 makedepends="gtk+3-devel libglib-devel openssl-devel gstreamer1-devel
  cairo-devel gettext-devel gst-plugins-base1-devel gst-plugins-bad1-devel
  gst1-editing-services-devel gtksourceview4-devel pango-devel gspell-devel
- gmp-devel libsecret-devel libhandy-devel rust-std"
+ gmp-devel libsecret-devel libhandy-devel rust-std gst1-editing-services-devel python3-devel cmake"
```
Comment (meator) [inline srcpkgs/fractal/template:11]: Why is `cmake` needed? It builds fine without it for me.

Comment (SpidFightFR) [reply srcpkgs/fractal/template:11]: A bunch of packages needed these... that's weird, that was the case for bijiben as well. Welp imma remove this cmake.

Comment (SpidFightFR) [reply srcpkgs/fractal/template]: there, it's now gone.

Comment (eli-schwartz) [reply srcpkgs/fractal/template]: What does that mean, "needed these"?

Projects with meson.build don't even use cmake at all unless the project does dependency('some-cmake-project'), in which case meson executes a small cmake dummy project and parses the cmake debug trace to see what cmake detected via a *-config.cmake; this is an alternative to pkg-config.

Meson will almost always print that it is checking for cmake as a system command, if it doesn't find an optional dependency. The fact that it prints "NO" doesn't mean that cmake is a dependency and should be installed as makedepends... it is just an informational message that cmake is not installed.

The actual error, if an error occurs, is always "some-package could not be found, tried pkg-config and cmake" and the solution is to install the actual dependency, not to try installing cmake.

Comment (eli-schwartz) [reply srcpkgs/fractal/template]: (note that installing cmake won't even help, in most cases.)

Comment (SpidFightFR) [reply srcpkgs/fractal/template]: > What does that mean, "needed these"?
> 
> Projects with meson.build don't even use cmake at all unless the project does dependency('some-cmake-project'), in which case meson executes a small cmake dummy project and parses the cmake debug trace to see what cmake detected via a *-config.cmake; this is an alternative to pkg-config.
> 
> Meson will almost always print that it is checking for cmake as a system command, if it doesn't find an optional dependency. The fact that it prints "NO" doesn't mean that cmake is a dependency and should be installed as makedepends... it is just an informational message that cmake is not installed.
> 
> The actual error, if an error occurs, is always "some-package could not be found, tried pkg-config and cmake" and the solution is to install the actual dependency, not to try installing cmake.

when i ran the testings at some point the checks returned an error with cmake labeled as missing... But it doesn't happen  anymore. I'm not actually sure of what's going on.

But if it works without, well it's good for me.

**Comment by SpidFightFR** [comment]
@ahesford hey, sorry for the ping, i hope you're doing great.

Do we have an update on this, by any chance ? :sweat_smile: 
Do i need to do something ? If so, don't hesitate to tell me ! :smile:

**Pull request merged**